index.js

```diff
@@ -41,13 +47,23 @@ class Replicate {
    * @param {string} options.userAgent - Identifier of your app
    * @param {string} [options.baseUrl] - Defaults to https://api.replicate.com/v1
    * @param {Function} [options.fetch] - Fetch function to use. Defaults to `globalThis.fetch`
+   * @param {Function} [options.prepareInput] - Function to prepare input data before sending it to the API.
    */
   constructor(options = {}) {
     this.auth = options.auth || process.env.REPLICATE_API_TOKEN;
     this.userAgent =
       options.userAgent || `replicate-javascript/${packageJSON.version}`;
     this.baseUrl = options.baseUrl || "https://api.replicate.com/v1";
     this.fetch = options.fetch || globalThis.fetch;
+    this.prepareInputs =
```

Comment:
I think there is value in providing customization. Though I'm not sure we need it immediately. 

I also think this isn't the right level of granularity. As implemented this doesn't provide any additional benefit to the user. If we expose this then the user will have to implement their own traversal + encoding logic and they could do this before passing the input into `run()`.

Instead we could provide something that calls a callback with a blob:

```ts
import * as Replicate from "replicate";

const replicate = new Replicate({
  async prepareInput(file: Blob | File | Buffer): string {
     // Replicate.encodeDataURI(file);
     // (await replicate.files.create(file)).urls.get;
  }
})
```

Or it might be enough to just provide a couple of strategies:

```js
const replicate = new Replicate({
  transformInputs: "default" | "embed" | "upload" | "none"
});
```

I like having `default` flexible, right now it will upload but fallback to embedding. In future we might decide assets smaller than a certain size should always be embedded etc.

Reply:
> they could do this before passing the input into run().

Oh, duh. That's the best argument against the proposed design — the user _already has complete control over input_. 

In that case, I quite like the idea of an enumerated parameter. Maybe `fileEncodingStrategy?: "default" | "upload" | "dataURI"`?